reconcile/ocm_subscription_labels/integration.py

```diff
@@ -0,0 +1,251 @@
+                    env=env.env,
+                    ocm_api=self.ocm_apis[env.env.name],
+                    cluster_details=cluster_details,
+                    labels=filtered_labels,
+                )
+        return states
+
+    def fetch_desired_state(self, clusters: Iterable[ClusterV1]) -> ClusterStates:
+        states: ClusterStates = {}
+        for cluster in clusters:
+            if cluster.ocm is None:
+                # already filtered out in get_clusters - make mypy happy
+                continue
+            states[cluster.name] = ClusterLabelState(
+                env=cluster.ocm.environment,
+                ocm_api=self.ocm_apis[cluster.ocm.environment.name],
+                labels=flatten(cluster.ocm_subscription_labels or {}),
+            )
+
+        return states
+
+    def reconcile(
+        self,
+        dry_run: bool,
+        current_cluster_states: ClusterStates,
+        desired_cluster_states: ClusterStates,
+    ) -> None:
+        for cluster_name, desired_cluster_state in desired_cluster_states.items():
+            try:
+                current_cluster_state = current_cluster_states[cluster_name]
+                if not (cluster_details := current_cluster_state.cluster_details):
+                    # this should never happen - make mypy happy
+                    raise RuntimeError("Cluster details not found.")
+
+                if desired_cluster_state == current_cluster_state:
+                    continue
+            except KeyError:
+                logging.info(
+                    f"Cluster '{cluster_name}' not found in OCM. Maybe it doesn't exist yet. Skipping."
+                )
+                continue
+
+            diff_result = diff_mappings(
+                current_cluster_state.labels, desired_cluster_state.labels
+            )
+
+            for label_to_add, value in diff_result.add.items():
+                logging.info(
+                    [
+                        "create_cluster_subscription_label",
+                        cluster_name,
+                        f"{label_to_add}={value}",
+                    ]
+                )
+                if not dry_run:
+                    add_subscription_label(
+                        ocm_api=desired_cluster_state.ocm_api,
+                        ocm_cluster=cluster_details.ocm_cluster,
+                        label=label_to_add,
+                        value=value,
+                    )
+            for label_to_rm, value in diff_result.delete.items():
+                logging.info(
+                    [
+                        "delete_cluster_subscription_label",
+                        cluster_name,
+                        f"{label_to_rm}={value}",
+                    ]
+                )
+                if not dry_run:
+                    delete_ocm_label(
+                        ocm_api=desired_cluster_state.ocm_api,
+                        ocm_label=cluster_details.labels[label_to_rm],
```

Comment:
here we are passing `cluster_details.labels[label_to_rm]`, is there a case that `cluster_details.labels[label_to_rm]` is not same as label value in diff? We are logging
```
                logging.info(
                    [
                        "delete_cluster_subscription_label",
                        cluster_name,
                        f"{label_to_rm}={value}",
                    ]
                )
```

Reply:
`cluster_details.labels[label_to_rm]` is the `OCMLabel` representation and provides the label URL. The label name is the key and this code is correct. No worries.